src/scene/sprite-atlas.js

```diff
@@ -0,0 +1,34 @@
+pc.extend(pc, function () {
+    'use strict';
+
+    /**
+    * @name pc.SpriteAtlas
+    * @class Represents the resource of a sprite atlas asset.
+    * @property {pc.Texture} texture The texture atlas.
+    * @property {Object} frames Contains frames which define portions of the texture atlas.
+    * @example
+    * var atlas = new pc.SpriteAtlas();
+    * atlas.frames = {
```

Comment:
This may be a dumb question, but why is this an object and not an array?

Reply:
Frames have a unique key and if you add or remove frames in between of other frames in the atlas the keys of the existing frames shouldn't change. So this is an easier way to change the atlas without having to change all the sprite assets that reference these frames.

Reply:
I recently noticed that for...in loops can be quite slow, so maybe some other solution (emulate linked list with 2 arrays?) will be better

Reply:
We don't loop through this object. The sprites assets reference keys of this object directly.